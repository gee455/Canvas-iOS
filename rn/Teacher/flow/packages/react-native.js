//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

declare var console: typeof console & { disableYellowBox: boolean }

export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}

export type KeyboardEventData = {
  endCoordinates: {
    width: number,
    height: number,
    screenX: number,
    screenY: number,
  },
}
