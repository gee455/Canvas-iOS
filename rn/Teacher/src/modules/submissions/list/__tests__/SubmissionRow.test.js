//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

/* eslint-disable flowtype/require-valid-file-annotation */

import 'react-native'
import React from 'react'
import SubmissionRow from '../SubmissionRow'
import type {
  SubmissionDataProps,
  GradeProp,
} from '../submission-prop-types'
import explore from '../../../../../test/helpers/explore'
import renderer from 'react-test-renderer'

jest
  .mock('TouchableHighlight', () => 'TouchableHighlight')
  .mock('../../../../common/components/Avatar', () => 'Avatar')

const mockSubmission = (status: SubmissionStatus = 'none', grade: ?GradeProp = null): SubmissionDataProps => {
  return {
    userID: '1',
    avatarURL: 'https://cats.pajamas/',
    name: 'Green Latern',
    status,
    grade,
    submissionID: null,
    submission: null,
    anonymous: false,
  }
}

test('unsubmitted ungraded row renders correctly', () => {
  const submission = mockSubmission()

  let tree = renderer.create(
    <SubmissionRow {...submission} onPress={jest.fn()} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('missing ungraded row renders correctly', () => {
  const submission = mockSubmission('missing')
  let tree = renderer.create(
    <SubmissionRow {...submission} onPress={jest.fn()} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('late graded row renders correctly', () => {
  const submission = mockSubmission('late', 'B-')
  let tree = renderer.create(
    <SubmissionRow {...submission} onPress={jest.fn()} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('submitted ungraded row renders correctly', () => {
  const submission = mockSubmission('submitted', 'ungraded')
  let tree = renderer.create(
    <SubmissionRow {...submission} onPress={jest.fn()} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('submitted not_graded row renders correctly', () => {
  const submission = mockSubmission('submitted', 'ungraded')
  submission.gradingType = 'not_graded'
  let tree = renderer.create(
    <SubmissionRow {...submission} onPress={jest.fn()} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('excused row renders correctly', () => {
  const submission = mockSubmission('missing', 'excused')
  let tree = renderer.create(
    <SubmissionRow {...submission} onPress={jest.fn()} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('onPress called on tap', () => {
  const submission = mockSubmission()
  const onPress = jest.fn()
  let row = explore(renderer.create(
    <SubmissionRow {...submission} onPress={onPress} />
  ).toJSON()).selectByID(`submission-${submission.userID}`)
  row && row.props.onPress()
  expect(onPress).toHaveBeenCalledWith(submission.userID)
})

test('anonymous grading doesnt show users names', () => {
  const submission = mockSubmission()
  let tree = renderer.create(
    <SubmissionRow {...submission} anonymous />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('pressing the avatar calls onAvatarPress', () => {
  const submission = mockSubmission()
  const onAvatarPress = jest.fn()
  let tree = renderer.create(
    <SubmissionRow {...submission} onAvatarPress={onAvatarPress} />
  ).toJSON()
  let avatar = explore(tree).selectByType('Avatar')
  avatar.props.onPress()
  expect(onAvatarPress).toHaveBeenCalledWith('1')
})
