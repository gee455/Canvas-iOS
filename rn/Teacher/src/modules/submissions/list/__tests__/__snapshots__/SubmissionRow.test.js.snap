// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`anonymous grading doesnt show users names 1`] = `
<View>
  <TouchableHighlight
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission-1"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "marginRight": 8,
            "width": 40,
          }
        }
      >
        <Avatar
          avatarURL={null}
          onPress={undefined}
          userName="Student"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
                "fontWeight": "600",
              },
              Object {
                "fontFamily": ".SFUIDisplay-semibold",
              },
            ]
          }
        >
          Student
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "fontSize": 14,
                  "paddingTop": 2,
                },
                undefined,
                Object {
                  "color": "#8B969E",
                },
              ],
              Object {
                "fontFamily": ".SFUIDisplay",
              },
            ]
          }
        >
          Not Submitted
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "#c7c7cc",
            "borderRightWidth": 2,
            "borderTopWidth": 2,
            "height": 10,
            "marginLeft": 7,
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
  </TouchableHighlight>
</View>
`;

exports[`excused row renders correctly 1`] = `
<View>
  <TouchableHighlight
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission-1"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "marginRight": 8,
            "width": 40,
          }
        }
      >
        <Avatar
          avatarURL="https://cats.pajamas/"
          onPress={undefined}
          userName="Green Latern"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
                "fontWeight": "600",
              },
              Object {
                "fontFamily": ".SFUIDisplay-semibold",
              },
            ]
          }
        >
          Green Latern
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "fontSize": 14,
                  "paddingTop": 2,
                },
                undefined,
                Object {
                  "color": "#EE0612",
                },
              ],
              Object {
                "fontFamily": ".SFUIDisplay",
              },
            ]
          }
        >
          Missing
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#374A59",
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {
                "alignSelf": "center",
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay-semibold",
            },
          ]
        }
      >
        Excused
      </Text>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "#c7c7cc",
            "borderRightWidth": 2,
            "borderTopWidth": 2,
            "height": 10,
            "marginLeft": 7,
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
  </TouchableHighlight>
</View>
`;

exports[`late graded row renders correctly 1`] = `
<View>
  <TouchableHighlight
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission-1"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "marginRight": 8,
            "width": 40,
          }
        }
      >
        <Avatar
          avatarURL="https://cats.pajamas/"
          onPress={undefined}
          userName="Green Latern"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
                "fontWeight": "600",
              },
              Object {
                "fontFamily": ".SFUIDisplay-semibold",
              },
            ]
          }
        >
          Green Latern
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "fontSize": 14,
                  "paddingTop": 2,
                },
                undefined,
                Object {
                  "color": "#FC5E13",
                },
              ],
              Object {
                "fontFamily": ".SFUIDisplay",
              },
            ]
          }
        >
          Late
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#374A59",
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {
                "alignSelf": "center",
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay-semibold",
            },
          ]
        }
      >
        B-
      </Text>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "#c7c7cc",
            "borderRightWidth": 2,
            "borderTopWidth": 2,
            "height": 10,
            "marginLeft": 7,
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
  </TouchableHighlight>
</View>
`;

exports[`missing ungraded row renders correctly 1`] = `
<View>
  <TouchableHighlight
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission-1"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "marginRight": 8,
            "width": 40,
          }
        }
      >
        <Avatar
          avatarURL="https://cats.pajamas/"
          onPress={undefined}
          userName="Green Latern"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
                "fontWeight": "600",
              },
              Object {
                "fontFamily": ".SFUIDisplay-semibold",
              },
            ]
          }
        >
          Green Latern
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "fontSize": 14,
                  "paddingTop": 2,
                },
                undefined,
                Object {
                  "color": "#EE0612",
                },
              ],
              Object {
                "fontFamily": ".SFUIDisplay",
              },
            ]
          }
        >
          Missing
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "#c7c7cc",
            "borderRightWidth": 2,
            "borderTopWidth": 2,
            "height": 10,
            "marginLeft": 7,
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
  </TouchableHighlight>
</View>
`;

exports[`submitted not_graded row renders correctly 1`] = `
<View>
  <TouchableHighlight
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission-1"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "marginRight": 8,
            "width": 40,
          }
        }
      >
        <Avatar
          avatarURL="https://cats.pajamas/"
          onPress={undefined}
          userName="Green Latern"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
                "fontWeight": "600",
              },
              Object {
                "fontFamily": ".SFUIDisplay-semibold",
              },
            ]
          }
        >
          Green Latern
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "fontSize": 14,
                  "paddingTop": 2,
                },
                undefined,
                Object {
                  "color": "#07AF1F",
                },
              ],
              Object {
                "fontFamily": ".SFUIDisplay",
              },
            ]
          }
        >
          Submitted
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "#c7c7cc",
            "borderRightWidth": 2,
            "borderTopWidth": 2,
            "height": 10,
            "marginLeft": 7,
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
  </TouchableHighlight>
</View>
`;

exports[`submitted ungraded row renders correctly 1`] = `
<View>
  <TouchableHighlight
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission-1"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "marginRight": 8,
            "width": 40,
          }
        }
      >
        <Avatar
          avatarURL="https://cats.pajamas/"
          onPress={undefined}
          userName="Green Latern"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
                "fontWeight": "600",
              },
              Object {
                "fontFamily": ".SFUIDisplay-semibold",
              },
            ]
          }
        >
          Green Latern
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "fontSize": 14,
                  "paddingTop": 2,
                },
                undefined,
                Object {
                  "color": "#07AF1F",
                },
              ],
              Object {
                "fontFamily": ".SFUIDisplay",
              },
            ]
          }
        >
          Submitted
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          color="#368BD8"
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 9,
                  "borderWidth": 1,
                  "fontSize": 11,
                  "fontWeight": "500",
                  "overflow": "hidden",
                  "paddingBottom": 0,
                  "paddingLeft": 6,
                  "paddingRight": 6,
                  "paddingTop": 3,
                  "textAlign": "center",
                },
                Object {
                  "borderColor": "#368BD8",
                  "color": "#368BD8",
                },
                Object {
                  "alignSelf": "flex-start",
                  "marginTop": 8,
                },
              ],
              Object {
                "fontFamily": ".SFUIDisplay-medium",
              },
            ]
          }
        >
          NEEDS GRADING
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "#c7c7cc",
            "borderRightWidth": 2,
            "borderTopWidth": 2,
            "height": 10,
            "marginLeft": 7,
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
  </TouchableHighlight>
</View>
`;

exports[`unsubmitted ungraded row renders correctly 1`] = `
<View>
  <TouchableHighlight
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission-1"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "marginRight": 8,
            "width": 40,
          }
        }
      >
        <Avatar
          avatarURL="https://cats.pajamas/"
          onPress={undefined}
          userName="Green Latern"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
                "fontWeight": "600",
              },
              Object {
                "fontFamily": ".SFUIDisplay-semibold",
              },
            ]
          }
        >
          Green Latern
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "fontSize": 14,
                  "paddingTop": 2,
                },
                undefined,
                Object {
                  "color": "#8B969E",
                },
              ],
              Object {
                "fontFamily": ".SFUIDisplay",
              },
            ]
          }
        >
          Not Submitted
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "#c7c7cc",
            "borderRightWidth": 2,
            "borderTopWidth": 2,
            "height": 10,
            "marginLeft": 7,
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
  </TouchableHighlight>
</View>
`;
